src/lib/molecules/InteractiveWidget/InteractiveWidget.tsx

```diff
@@ -0,0 +1,187 @@
+     * Possible values: `'default' | 'compact'`
+     */
+    appearance?: 'default' | 'compact';
+    /**
+     * The `iconColor` prop determines the color of the icon in the component.
+     * It should be a string representing a valid CSS color value.
+     */
+    iconColor?: string;
+    /**
+     * Label for tag component.
+     * Adding props you will add the `<Tag/>` component.
+     */
+    tagName?: string;
+    /**
+     * color for tag.
+     */
+    tagColor?: string;
+    /**
+     * Props for `<Switcher/>` component (see `<Switcher/>` component documentation).
+     * Adding props you wil add the component.
+     */
+    switcherProps?: typeof Switcher.propTypes;
+    /**
+     * The main descriptive text or title for the component.
+     */
+    title?: string;
+    /**
+     * Additional information or tooltip string that appears when hovering over the info icon next to the title.
+     * If not provided, the info icon will not be displayed.
+     */
+    titleInfo?: string;
+    /**
+     * Description paragraph.
+     */
+    description?: string;
+    /**
+     * icon background for default appearance.
+     */
+    iconBackground?: boolean;
+}
+
+const InteractiveWidget: FC<IInteractiveWidgetProps> = ({
+    className,
+    withBorder = true,
+    onClick,
+    disabled = false,
+    icon,
+    title,
+    appearance = 'default',
+    iconColor,
+    tagName,
+    tagColor,
+    switcherProps,
+    titleInfo,
+    description,
+    iconBackground = true,
+    ...restProps
+}) => {
+    const titleRef = useRef<HTMLHeadingElement | null>(null);
+    const descriptionRef = useRef<HTMLParagraphElement | null>(null);
+    const isTitleTruncated = useEllipsisDetection(titleRef);
+    const isDescriptionTruncated = useEllipsisDetection(descriptionRef);
+
+    return (
+        <div
+            className={classNames(
+                'interactiveWidget',
+                {
+                    'interactiveWidget-border': withBorder,
+                    'interactiveWidget-pointer': onClick,
+                    'interactiveWidget-disabled': disabled
+                },
+                className
+            )}
+            {...(!disabled && onClick ? { onClick: (event: MouseEvent<HTMLDivElement>) => onClick(event) } : {})}
+            tabIndex={0}
+            onKeyDown={(event: KeyboardEvent<HTMLDivElement>) => {
+                onClick && onClick(event);
+            }}
+            {...restProps}
+        >
+            <div className={'interactiveWidget__wrapper'}>
+                {icon && appearance === 'default' && (
+                    <InteractiveWidgetIcon
+                        children={icon}
+                        appearance={appearance}
+                        iconColor={iconColor}
+                        iconBackground={iconBackground}
+                    />
+                )}
+                <div className={classNames('interactiveWidget__content')}>
```

Comment:
Why do you need the `classNames` function here? 
use it when you have dynamic combinations.